src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2259,6 +2259,18 @@ private BoundExpression BindExplicitNullableCastFromNonNullable(ExpressionSyntax
             {
                 bag.Free();
             }
+
+            ErrorCode GetErrorCode(ConstantValue constantValue)
+            {
+                if (constantValue == ConstantValue.Bad)
```

Comment:
>constantValue == ConstantValue.Bad [](start = 20, length = 34)

This feels fragile, who knows why the constant expression is bad and whether it is caused by come child expression. I think we would want BindCastCore to return explicitly whether there was an overflow error and based on that we can decide what error to report.